### PR TITLE
Localization: add base_map files in pyramid_map module

### PR DESCRIPTION
**Pull request by tthhee**

### CODE REVIEW

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+class BaseMapMatrixHandler {
+ public:
+  BaseMapMatrixHandler() {}
+  virtual ~BaseMapMatrixHandler() {}
+  /**@brief Load the map cell from a binary chunk.
+   * @param <return> The size read (the real size of object).
+   */
+  virtual size_t LoadBinary(const unsigned char* buf,
+                                       BaseMapMatrix* matrix) = 0;
```
Comment (jinghaomiao) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h]: Align your parameters across lines.

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.cc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+BaseMapMatrix::BaseMapMatrix() {}
+
+BaseMapMatrix::~BaseMapMatrix() {}
+
+BaseMapMatrix::BaseMapMatrix(const BaseMapMatrix& map_matrix) {}
+
+bool BaseMapMatrix::GetIntensityImg(cv::Mat* intensity_img) const {
+    return false;
+}
+  /**@brief get altitude image of node. */
```
Comment (michael4338) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.cc]: indent

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  /**@brief The default constructor. */
+  BaseMapMatrix();
+  /**@brief The deconstructor. */
+  virtual ~BaseMapMatrix();
+  /**@brief The copy constructor. */
+  explicit BaseMapMatrix(const BaseMapMatrix& map_matrix);
+  /**@brief Initialize the map matrix. */
+  virtual void Init(const BaseMapConfig& config) = 0;
+  /**@brief Reset map cells data. */
+  //    virtual void Reset(const BaseMapConfig* config) = 0;
```
Comment (michael4338) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h]: space or remove unused code

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+#pragma once
+
+#include <assert.h>
+#include <vector>
+#include "modules/localization/msf/local_map/base_map/base_map_fwd.h"
+#include "opencv2/opencv.hpp"
```
Comment (michael4338) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h]: incorrect include sequence

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  /**@brief Create the binary. Serialization of the object.
+   * @param <buf, buf_size> The buffer and its size.
+   * @param <return> The required or the used size of is returned.
+   */
+  virtual size_t CreateBinary(const BaseMapMatrix* matrix,
+                                         unsigned char* buf,
```
Comment (michael4338) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h]: align parameters

File: modules/localization/msf/local_pyramid_map/base_map/base_map_node_index.h [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                                  const MapNodeIndex& index);
+
+  /**@brief The ID of the resolution.
+   * Should be less than BaseMapConfig::_map_resolutions.size(). */
+  unsigned int resolution_id_ = 0;
+  /**@brief The zone ID. 1 - 60 and -1 - -60.
+   * The positive value is the zone at the north hemisphere. */
+  int zone_id_ = 10;
+  /**@brief The map node ID at the northing direction. */
+  unsigned int m_ = 0;
```
Comment (michael4338) [inline modules/localization/msf/local_pyramid_map/base_map/base_map_node_index.h]: more meaningful name?

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h (hunk 1 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h]: fixed

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.cc (hunk 2 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.cc]: fixed

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h (hunk 3 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h]: fixed

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h (hunk 4 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_matrix.h]: fixed

File: modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h (hunk 5 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_matrix_handler.h]: fixed

File: modules/localization/msf/local_pyramid_map/base_map/base_map_node_index.h (hunk 6 above)
Comment (tthhee) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_node_index.h]: we use "m_" in every version of the map code, so to keep it consistent, is it better not to change it? besides, we have comments.

Comment (michael4338) [reply modules/localization/msf/local_pyramid_map/base_map/base_map_node_index.h]: fair enough

**Pull request merged**